### PR TITLE
Re-branding app menu's icon

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/38589

image
image
image
Screenshot 2024-06-05 at 10 45 32 AM
Screenshot 2024-06-05 at 10 45 38 AM
Screenshot 2024-06-05 at 9 59 33 PM

Resolves

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See the linked issue

### CODE REVIEW

**Comment by simonhong** [comment]
We have another new sub menus - `Find and Edit` and `Save and Share`
I think we need to update the icons of their sub menu entries.
`Find` is our icon but others are from upstream.
Screenshot 2024-06-04 at 12 10 49 PM

image


and if we want to enable `Reading List` sub menu, their ones also needs to be updated.
Screenshot 2024-06-04 at 12 05 30 PM

**Comment by aguscruiz** [comment]
@simonhong added all the missing ones to our Figma 
https://www.figma.com/design/H11ZOl6JMYbCTW4ZJXqR5V/%F0%9F%A6%81-Browser?node-id=6697-27337&t=hv9qaLCoqTHbRLip-1

There's a couple new icons. that need to be synced from Nala, like "Cut"

**Comment by simonhong** [comment]
> @simonhong added all the missing ones to our Figma https://www.figma.com/design/H11ZOl6JMYbCTW4ZJXqR5V/%F0%9F%A6%81-Browser?node-id=6697-27337&t=hv9qaLCoqTHbRLip-1
> 
> There's a couple new icons. that need to be synced from Nala, like "Cut"

@aguscruiz Just want to double check about `Install facebook` icon above.
figma uses clipboard icon for it. it's right?

File: browser/ui/toolbar/brave_app_menu_model.cc [hunk 1]
```diff
@@ -198,9 +198,25 @@ void BraveAppMenuModel::Build() {
 
   ApplyLeoIcons(this);
   ApplyLeoIcons(bookmark_sub_menu_model());
+
   for (const auto& submenu : sub_menus()) {
     ApplyLeoIcons(submenu.get());
   }
+
+  // TODO(simonhong): apply reo icons recursively.
```
Comment (fallaciousreasoning) [inline browser/ui/toolbar/brave_app_menu_model.cc:206]: mega nit: 
```suggestion
  // TODO(simonhong): apply Nala icons recursively.
```

**Review by fallaciousreasoning — APPROVED** [review]
Looks great, thanks @simonhong !

**Comment by aguscruiz** [comment]
> > @simonhong added all the missing ones to our Figma https://www.figma.com/design/H11ZOl6JMYbCTW4ZJXqR5V/%F0%9F%A6%81-Browser?node-id=6697-27337&t=hv9qaLCoqTHbRLip-1
> > There's a couple new icons. that need to be synced from Nala, like "Cut"
> 
> @aguscruiz Just want to double check about `Install facebook` icon above. figma uses clipboard icon for it. it's right? We're using another icon so far. image

Whoops, my bad. Yeah the one you have is the right one. I'll fix in Figma
image

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.68.76 Chromium: 126.0.6478.26 (Official Build) nightly (64-bit)
-- | --
Revision | 997c487ad3e5dee61238fe8fed55bb085aa89074
OS | Windows 11 Version 23H2 (Build 22631.3672)
```

Using the STR/examples outlined via https://github.com/brave/brave-browser/issues/38589#issue-2318458272, https://github.com/brave/brave-browser/issues/38589#issuecomment-2133655960 & https://github.com/brave/brave-core/pull/23976#issue-2332427962, ensured that the correct icons were being used via the `Hamburger Menu` as per the following:

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`**
----------------|----------------|----------------|----------------|----------------|-----------------
image | image | image | image | image | image

**Pull request merged**